### ISSUE TITLE
PEP 517/518 compliance and pyproject.toml

### ISSUE THREAD

**Issue by ChrisDaunt1984** [Nubis-Communications/SignalIntegrity#84]
When I'm installing it in editable mode (pip install -e .) with our current Python 3.12 and recent pip versions, I'm getting this deprecation warning:

```bash
DEPRECATION: Legacy editable install of SignalIntegrity==1.4.1 
from file:///... (setup.py develop) is deprecated. pip 25.3 will enforce this behaviour change. 
A possible replacement is to add a pyproject.toml or enable --use-pep517, and use setuptools >= 64.
```

While it still installs, this means pip (and other modern packaging tools) are flagging setup.py-only projects. Eventually, this could stop working. python 3.12 makes some changes to distutils so seems to be getting flagged more now. 

The solution is to define metadata in the minimal pyproject.toml file according to PEP517 and PEP518
